package/MDAnalysis/analysis/msd.py

```diff
@@ -346,7 +346,8 @@ def _prepare(self):
                                                   self.n_particles))
         self._position_array = np.zeros(
             (self.n_frames, self.n_particles, self.dim_fac))
-        # self.results.timeseries not set here
+        # set the position array here to remove _single_frame from msd.py 
+        self.results.timeseries = self._position_array
```

Comment:
Obtain the coordinates via [trajectory.timeseries](https://docs.mdanalysis.org/stable/documentation_pages/coordinates/base.html#MDAnalysis.coordinates.base.ReaderBase.timeseries).